intro_to_R_tidyverse/intro_to_tidyverse.Rmd

```diff
@@ -0,0 +1,185 @@
+other files types.
+
+### About directories
+Before we can import the file we need though, we should double check what
+directory R is in, aka what our current *directory* is. 
+We can do this by using the `getwd()` function, which will tell us what folder
+we are in. 
+Then we can use the `dir()` function to have R list the files in our current 
+directory. 
+What we should see is the file `metadata.tsv` show up as one of the files when 
+we run `dir()`.
+If for some reason we were not in the correct directory, we can either set our
+directory to somewhere else using `setwd()` OR we can tell R to look for the 
+file `metadata.tsv` somewhere else using a relative `file.path()`.
+We will discuss and practice these concepts more in other modules as we go along.
+```{r}
+# Let's check what directory we are in:
+getwd()
+
+# Let's check what files are here
+dir()
+
+# The file we are reading in is our argument 
+metadata <- read_tsv("metadata.tsv")
+```
+
+## dplyr pipes
+A nifty thing with the tidyverse is the pipes: `%>%`
+This handy thing allows you to funnel the result of one expression to the next
+making your code a little more streamlined.
+
+For example, the output from this:  
+```
+filter(metadata, sex == "F")
+```  
+...is the same as the output from this:  
+```
+metadata %>% filter(sex == "F")
+```  
+  
+This makes it so your code is cleaner and easier to read. 
+Let's look at an example with our metadata of of how the same 
+functions look with or without pipes:
+
+*Example 1:* without pipes:   
+```{r}
+metadata_edit <- metadata
+metadata_edit <- arrange(metadata_edit, histology)
+metadata_edit <- filter(metadata_edit, follow_up_mths > 50)
+metadata_edit <- select(metadata_edit, sample_id, sex, organism)
+```
+UGH, we have to repeat and assign `special.mtcars` so many times here! 
+It's annoying and makes it harder for people to read. 
+
+*Example 2:* Same result as 1 but with pipes!
+```{r}
+# Example of the same modifications as above but with pipes!
+metadata_edit <- metadata %>%
+                 arrange(histology) %>%
+                 filter(follow_up_mths > 50) %>%
+                 select(sample_id, sex, organism)
+```
+
+Now that hopefully you are convinced that tidyverse helps your code to be neater and
+easier to use and read, let's go through some of the popular tidyverse functions
+and so we can create pipelines like this. 
+
+## Common tidyverse functions
+
+Let's say we wanted to filter this gene expression dataset to particular sample
+groups.
+In order to do this, we would use the function `filter` as well as a logic statement.
+```{r}
+# Here let's filter the data to have only females
+metadata %>% filter(sex == "F")
+```
+
+We can use filter similarly for numeric statements.  
+```{r}
+# Here let's filter the data to have people only older than 
+metadata %>% filter(age_mths > 100)
+```
+
+What if we wanted filter this dataset to males who are younger than 100 months? 
```

Comment:
I think this is filtering to _older_ than 100 months

Reply:
Whoops